### PR TITLE
Structure_Engine: Add support for Timber, Aluminium and Generic sections

### LINKED ISSUE #1409: Structure_Engine: Add support methods for Timber, Aluminium and Generic sections
#### Description:

Add methods to support new classes added in https://github.com/BHoM/BHoM/issues/555 and https://github.com/BHoM/BHoM/issues/564

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/654
   
### Issues addressed by this PR

Closes #1409 

### Test files

https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/BHoM/Structure_oM/Issue555-AddingTimberALuminiumAbdGenericSections?csf=1&e=HMfIUS

### Changelog

- Add Create methods for Aluminium, Timber and Generic sections
- Update Geometry and Extrude methods to work with new `IGeometricalSection` interface
- Adding support for new section types for converting to and From `IFramingElement`
- Adding descriptions for create methods for section properties

### CODE REVIEW

File: Structure_Engine/Create/SectionProperty.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                    return ConcreteSectionFromProfile(profile, material as Concrete, name);
+                case MaterialType.Aluminium:
+                    return AluminiumSectionFromProfile(profile, material as Aluminium, name);
+                case MaterialType.Timber:
+                    return TimberSectionFromProfile(profile, material as Timber, name);
+                case MaterialType.Rebar:
+                case MaterialType.Tendon:
+                case MaterialType.Glass:
+                case MaterialType.Cable:
+                case MaterialType.Undefined:
```
Comment (JosefTaylor) [inline Structure_Engine/Create/SectionProperty.cs]: Should generic materials, with `MaterialType.Undefined` also fall through to raise the warning?

Comment (IsakNaslundBh) [reply Structure_Engine/Create/SectionProperty.cs]: Hmm you might be right! Kind of wanted to keep warnings for whenever GenericSection was used as a highlight that you might want to do something else..

Comment (IsakNaslundBh) [reply Structure_Engine/Create/SectionProperty.cs]: Yes, thought about it as well, but in general I thought it was good to keep the warning as a reminder that you might want to use something else. Can remove later if to annoying?

**Pull request merged**